### PR TITLE
Macro-generated connectConnector methods.

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR is the counterpart to #1441 for connectors. The original plan was to rename connectors to sockets first before introducing the `connectSocket_*()` macro function. But we decided to switch the order of these changes, since having the macro-generated method in place for the alpha release is perhaps more important than having the rename to sockets for the alpha.

This is my second attempt at creating this PR, as #1446 shows an outdated diff (from before #1441 was merged).

### CODE REVIEW

File: OpenSim/Common/ComponentConnector.h [hunk 1]
```diff
@@ -857,7 +859,14 @@ class Input : public AbstractInput {
                 "Path to a Component to satisfy the Connector '"            \
                 #cname "' of type " #T " (description: " comment ").")      \
     };                                                                      \
-    /** @endcond                                                         */
+    /** @endcond                                                         */ \
```
Comment (aseth1) [inline OpenSim/Common/ComponentConnector.h]: Update auto generated comment for the property to : "Path to a Component that satisfies the Connector ` ... "

**Comment by chrisdembia** [comment]
@aseth1 I have addressed your comment.

**Comment by chrisdembia** [comment]
Ready for review.

**Comment by aseth1** [comment]
LGTN :+1:   @aymanhab?

**Comment by aymanhab** [comment]
LGTM as well. Thanks @chrisdembia

**Comment by chrisdembia** [comment]
Thank you.

**Pull request merged**